### DISCUSSION TITLE
Geting mwp working with ELRS CRSF telemetry from a BT USB Adapter

### DISCUSSION THREAD

**Discussion by Pelado-Mat** [stronnag/mwptools#172] [Q&A] [answered]
Im currently using  [Telemetry APP](https://github.com/RomanLut/android-taranis-smartport-telemetry) to display telemetry on my phone, but I would like to use my PC too. 
My setup is a RM Zorro ELRS with a non-BLE BT Module on AUX 1 configured as telemetry mirror. It works fine with the app.

I'm trying to use mwp with the same setup, but the telemetry never gets into mwp:
```
2024-04-10T15:55:59-0400 Try connect 0B:3A:0D:12:54:45 RM Zorro@115200
2024-04-10T15:55:59-0400 Connected 0B:3A:0D:12:54:45 RM Zorro@115200
2024-04-10T15:55:59-0400 Serial ready
2024-04-10T15:55:59-0400 Comm error count 34
2024-04-10T15:55:59-0400 expected header0 (ea)
2024-04-10T15:55:59-0400 Comm error count 35
2024-04-10T15:55:59-0400 MAVCRC Fail, got 64e3 != f805 (cmd=234, len=119)
2024-04-10T15:56:00-0400 MSP Timeout (MSP_CMDS_IDENT)
2024-04-10T15:56:01-0400 MSP Timeout (MSP_CMDS_IDENT)
2024-04-10T15:56:02-0400 MSP Timeout (MSP_CMDS_IDENT)
2024-04-10T15:56:03-0400 MSP Timeout (MSP_CMDS_IDENT)
2024-04-10T15:56:04-0400 MSP Timeout (MSP_CMDS_IDENT)
2024-04-10T15:56:05-0400 MSP Timeout (MSP_CMDS_IDENT)
2024-04-10T15:56:06-0400 MSP Timeout (MSP_CMDS_IDENT)
2024-04-10T15:56:06-0400 Comm error count 36
2024-04-10T15:56:06-0400 fail on header1 ea
2024-04-10T15:56:07-0400 MSP Timeout (MSP_CMDS_IDENT)
2024-04-10T15:56:08-0400 MSP Timeout (MSP_CMDS_IDENT)
2024-04-10T15:56:09-0400 Comm error count 37
2024-04-10T15:56:09-0400 MAVCRC Fail, got dea != 6fc5 (cmd=238, len=234)
2024-04-10T15:56:09-0400 MSP Timeout (MSP_CMDS_IDENT)
2024-04-10T15:56:10-0400 MSP Timeout (MSP_CMDS_IDENT)
2024-04-10T15:56:10-0400 Comm error count 38
2024-04-10T15:56:10-0400 fail on header1 ea
```
I tried to follow the online manual, but can't find how to force CRSF telemety.

**Comment by stronnag** [comment]
I have a RM Boxer here with ELRS and it works OK, using EdgeTX TELEM Mirror and the Boxer USB port. 
Telem mirror also works with a TX16S, BT and either Frsky or IBUS.

Can you please capture the raw data by running MWP from the command line as:
```
mwp --raw-log # and anything else you wish
```

This will create a file in the current directory, for example `mwp._dev_ttyACM0.2024-04-10T210403.raw` (i.e some combination of `mwp`, device name and date, `.raw`.

Post that file here, together with the **full** `mwp_stderr` file (not some subset).

**Comment by Pelado-Mat** [comment]
Hope it helps.
Im running EdgeTX 2.10rc3, the BT Module is working an 115200 bauds

[mwp_stderr.txt](https://github.com/stronnag/mwptools/files/14937364/mwp_stderr.txt)
[mwp_2024-04-10_161556.raw.txt](https://github.com/stronnag/mwptools/files/14937365/mwp_2024-04-10_161556.raw.txt)
[mwp_2024-04-10_161547.log](https://github.com/stronnag/mwptools/files/14937366/mwp_2024-04-10_161547.log)
[mwp_2024-04-10_161533.raw.txt](https://github.com/stronnag/mwptools/files/14937367/mwp_2024-04-10_161533.raw.txt)
[mwp_2024-04-10_161315.raw.txt](https://github.com/stronnag/mwptools/files/14937369/mwp_2024-04-10_161315.raw.txt)
[mwp_2024-04-10_161203.raw.txt](https://github.com/stronnag/mwptools/files/14937370/mwp_2024-04-10_161203.raw.txt)
[mwp_2024-04-10_161054.raw.txt](https://github.com/stronnag/mwptools/files/14937371/mwp_2024-04-10_161054.raw.txt)

**Comment by stronnag** [comment]
Final chance. A complete, unadulterated ` ~/mwp_stderr_2024-04-10.txt`. Otherwise I'm not prepared to play guessing names.

**Reply by Pelado-Mat** [reply]
Sorry, I did not get that the program created that file. Here is attached.

[mwp_stderr_2024-04-10.txt](https://github.com/stronnag/mwptools/files/14937461/mwp_stderr_2024-04-10.txt)

**Comment by stronnag** [comment]
Thank you. 
Could you also record some longer raw data? Like several minutes. And ensure there is more valid telemetry from the radio?

I ran all 14s of `mwp_2024-04-10_161054.raw.txt` through mwp's `crsfparser` tool,  see attached log: [crsf.log](https://github.com/stronnag/mwptools/files/14937568/crsf.log)

There's not much useful for mwp here, but attitude, RSSI and flight mode are recognised and (see below) displayed.

The log was then replayed into mwp (today's github  master branch (for which `--no-poll` works correctly)).
Having **first started** mwp as:
```
mwp -d udp://:3000 -a --no-poll
```
And selected the CRSF filter before replaying the logged data:
```
mwp-log-replay -d udp://localhost:3000  /home/jrh/dl/mwp_2024-04-10_161054.raw.txt
```
![image](https://github.com/stronnag/mwptools/assets/158229/3268ff7b-31d0-4ca8-9354-2274d538f983)

So as far as I can see:

* The data is decoded and displayed correctly.
* There's not such data to test
* There's little useful in data (no useful GPS or VBAT for example)
* Selecting the [CRSF filter]( https://www.daria.co.uk/mwptools/mwp-multi-procotol/#auto-detection) will help

**Comment by Pelado-Mat** [comment]
That's weird! on my PC I cannot select CRSF. I swa it on the docs but assumed it was an old screen capture

![image](https://github.com/stronnag/mwptools/assets/16164175/832c87a9-4f70-4a79-aa85-e205fbb2d186)


> ![image](https://private-user-images.githubusercontent.com/158229/321397262-3268ff7b-31d0-4ca8-9354-2274d538f983.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI3OTI2OTUsIm5iZiI6MTcxMjc5MjM5NSwicGF0aCI6Ii8xNTgyMjkvMzIxMzk3MjYyLTMyNjhmZjdiLTMxZDAtNGNhOC05MzU0LTIyNzRkNTM4Zjk4My5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDEwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQxMFQyMzM5NTVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jNGViOTFiMDc3ZmY3MjEzNzQzNGJjM2MzMzM4NzE1YzE5MjZjMDYzNGU5M2NjOTlkZjk0ODRiYmZlOWZkMGJiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.whcI4sji4Nemj5PgMkzfw8pAkvxtplls5PiCoZC2eDM)
> 
> So as far as I can see:
> 
>     * The data is decoded and displayed correctly.
> 
>     * There's not such data to test
> 
>     * There's little useful in data (no useful GPS or VBAT for example)
> 
>     * Selecting the [CRSF filter](https://www.daria.co.uk/mwptools/mwp-multi-procotol/#auto-detection) will help

**Comment by Pelado-Mat** [comment, marked as answer]
Well I find the problem.... I had installed in my ~/.local an really ancient mwptools version 4.x. I really do not remember when I did that...
```
mwp f23d52f / 2021-08-15T10:45:24+01:00 4.227.350 on Wayland 
```

I now remove it and reinstalled the .deb from the releases page. After fighting with dconf I finally could connect!

Sorry for the time spent and thanks for all the help.
